### DISCUSSION TITLE
adding modules to devices using pynetbox

### DISCUSSION THREAD

**Discussion by wannagofast83** [netbox-community/pynetbox#656] [General]
I have tried a number of ways to get a script to import modules and then adopt components and it fails every time because there is already a interface with that name on the device.

'nb.dcim.modules.create( device = nb_device.id, 
module_bay = nb_device_module_bay.id, 
module_type = nb_module_type.id, 
status = "active", 
serial = _module_serial, 
#"asset_tag": "",
#"description": "",
#"comments": "",
replicate_components = False,
 adopt_components= True,
 #"tags": "",
 #"id": "",
 )'

OR

'nb_device.adopt_components = True'

Anyone have any advice?

**Comment by Kani999** [comment]
@wannagofast83 Did you solve the problem?

**Reply by wannagofast83** [reply]
Sorry for the late reply. I didn't. I don't think that the API has the capability to adopt or replicate. 

[18028](https://github.com/netbox-community/netbox/discussions/18028) I created this and they created an issue for it. 
[Issue 21388](https://github.com/netbox-community/netbox/issues/21388)
